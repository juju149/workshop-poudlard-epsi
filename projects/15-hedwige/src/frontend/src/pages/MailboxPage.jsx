import React, { useState, useEffect } from 'react';
import { Link, useNavigate } from 'react-router-dom';
import EmailList from '../components/EmailList';
import EmailDetail from '../components/EmailDetail';
import '../styles/MailboxPage.css';

function MailboxPage({ sessionId, user, onLogout }) {
  const [emails, setEmails] = useState([]);
  const [selectedEmail, setSelectedEmail] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);
  const navigate = useNavigate();

  useEffect(() => {
    fetchEmails();
  }, [sessionId]);

  const fetchEmails = async () => {
    try {
      setLoading(true);
      setError(null);
      
      const response = await fetch('/api/emails?maxResults=20', {
        headers: {
          'x-session-id': sessionId
        }
      });

      const data = await response.json();

      if (response.ok) {
        setEmails(data.emails || []);
      } else {
        setError(data.error || 'Failed to fetch emails');
      }
    } catch (err) {
      setError('Failed to connect to server');
    } finally {
      setLoading(false);
    }
  };

  const handleEmailClick = async (emailId) => {
    try {
      const response = await fetch(`/api/emails/${emailId}`, {
        headers: {
          'x-session-id': sessionId
        }
      });

      const data = await response.json();

      if (response.ok) {
        setSelectedEmail(data);
      } else {
        setError(data.error || 'Failed to fetch email');
      }
    } catch (err) {
      setError('Failed to load email');
    }
  };

  const handleBackToList = () => {
    setSelectedEmail(null);
  };

  const handleCompose = () => {
    navigate('/compose');
  };

  return (
    <div className="mailbox-page">
      <header className="mailbox-header">
        <div className="header-left">
          <h1>🦉 Hedwige</h1>
        </div>
        <div className="header-right">
          {user && (
            <div className="user-info">
              {user.picture && <img src={user.picture} alt={user.name} />}
              <span>{user.name}</span>
            </div>
          )}
          <button onClick={onLogout} className="logout-btn">Logout</button>
        </div>
      </header>

      <div className="mailbox-content">
        <div className="mailbox-sidebar">
          <button className="compose-btn" onClick={handleCompose}>
            ✉️ Compose
          </button>
          <button className="refresh-btn" onClick={fetchEmails}>
            🔄 Refresh
          </button>
        </div>

        <div className="mailbox-main">
          {loading ? (
            <div className="loading">Loading emails...</div>
          ) : error ? (
            <div className="error">{error}</div>
          ) : selectedEmail ? (
            <EmailDetail 
              email={selectedEmail} 
              onBack={handleBackToList} 
            />
          ) : (
            <EmailList 
              emails={emails} 
              onEmailClick={handleEmailClick}
            />
          )}
        </div>
      </div>
    </div>
  );
}

export default MailboxPage;
