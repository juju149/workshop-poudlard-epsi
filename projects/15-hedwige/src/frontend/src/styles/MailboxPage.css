.mailbox-page {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  background: #f5f5f5;
}

.mailbox-header {
  background: white;
  padding: 15px 30px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.header-left h1 {
  font-size: 1.8em;
  color: #333;
}

.header-right {
  display: flex;
  align-items: center;
  gap: 20px;
}

.user-info {
  display: flex;
  align-items: center;
  gap: 10px;
}

.user-info img {
  width: 36px;
  height: 36px;
  border-radius: 50%;
}

.logout-btn {
  padding: 8px 16px;
  background: #667eea;
  color: white;
  border: none;
  border-radius: 6px;
  font-weight: 500;
  transition: background 0.3s;
}

.logout-btn:hover {
  background: #5568d3;
}

.mailbox-content {
  display: flex;
  flex: 1;
  overflow: hidden;
}

.mailbox-sidebar {
  width: 200px;
  background: white;
  padding: 20px;
  border-right: 1px solid #ddd;
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.compose-btn,
.refresh-btn {
  padding: 12px 20px;
  border: none;
  border-radius: 6px;
  font-weight: 500;
  transition: all 0.3s;
}

.compose-btn {
  background: #667eea;
  color: white;
}

.compose-btn:hover {
  background: #5568d3;
  transform: translateY(-1px);
  box-shadow: 0 4px 8px rgba(102, 126, 234, 0.3);
}

.refresh-btn {
  background: #f0f0f0;
  color: #333;
}

.refresh-btn:hover {
  background: #e0e0e0;
}

.mailbox-main {
  flex: 1;
  overflow-y: auto;
  background: white;
  margin: 20px;
  border-radius: 8px;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1);
}

.loading,
.error {
  padding: 40px;
  text-align: center;
  font-size: 1.2em;
  color: #666;
}

.error {
  color: #c33;
}
