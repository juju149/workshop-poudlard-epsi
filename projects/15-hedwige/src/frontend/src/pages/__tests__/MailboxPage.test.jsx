import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MailboxPage from '../MailboxPage';

describe('MailboxPage', () => {
  it('correspond au snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <MailboxPage />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
