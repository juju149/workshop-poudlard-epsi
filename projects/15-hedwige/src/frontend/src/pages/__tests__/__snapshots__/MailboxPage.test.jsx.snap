// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MailboxPage > correspond au snapshot 1`] = `
<DocumentFragment>
  <div
    class="mailbox-page"
  >
    <header
      class="mailbox-header"
    >
      <div
        class="header-left"
      >
        <h1>
          ü¶â Hedwige
        </h1>
      </div>
      <div
        class="header-right"
      >
        <button
          class="logout-btn"
        >
          Logout
        </button>
      </div>
    </header>
    <div
      class="mailbox-content"
    >
      <div
        class="mailbox-sidebar"
      >
        <button
          class="compose-btn"
        >
          ‚úâÔ∏è Compose
        </button>
        <div
          class="mailbox-tabs sidebar-tabs"
        >
          <button
            class="tab-active"
          >
            üì• Mail re√ßu
          </button>
          <button
            class=""
          >
            üì§ Mail envoy√©
          </button>
        </div>
      </div>
      <div
        class="mailbox-main"
      >
        <div
          class="email-list"
        >
          <h2>
            Inbox (2)
          </h2>
          <div
            class="email-items"
          >
            <div
              class="email-item"
            >
              <div
                class="email-item-header"
              >
                <span
                  class="email-from"
                >
                  Minerva McGonagall
                </span>
                <span
                  class="email-date"
                >
                  Fri
                </span>
              </div>
              <div
                class="email-subject"
              >
                Bienvenue √† Poudlard
              </div>
              <div
                class="email-snippet"
              />
            </div>
            <div
              class="email-item"
            >
              <div
                class="email-item-header"
              >
                <span
                  class="email-from"
                >
                  Hermione Granger
                </span>
                <span
                  class="email-date"
                >
                  Sat
                </span>
              </div>
              <div
                class="email-subject"
              >
                Devoirs
              </div>
              <div
                class="email-snippet"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
